masonry/src/widgets/flex.rs

```diff
@@ -768,31 +745,28 @@ impl Widget for Flex {
                     let child_size = ctx.child_size(widget);
                     let alignment = alignment.unwrap_or(self.cross_alignment);
                     let child_minor_offset = match alignment {
-                        // This will ignore baseline alignment if it is overridden on children,
-                        // but is not the default for the container. Is this okay?
-                        CrossAxisAlignment::Baseline
-                            if matches!(self.direction, Axis::Horizontal) =>
-                        {
+                        CrossAxisAlignment::Baseline if self.direction == Axis::Horizontal => {
+                            let max_height = max_below_baseline + max_above_baseline;
+                            let extra_height = (minor - max_height).max(0.);
+
                             let child_baseline = ctx.child_baseline_offset(widget);
                             let child_above_baseline = child_size.height - child_baseline;
                             extra_height + (max_above_baseline - child_above_baseline)
                         }
                         CrossAxisAlignment::Fill => {
                             let fill_size: Size = self
                                 .direction
-                                .pack(self.direction.major(child_size), minor_dim)
+                                .pack(self.direction.major(child_size), minor)
                                 .into();
-                            if ctx.old_size() != fill_size {
```

Comment:
That block was there for layout caching, but (1) we now do that at the Masonry level, (2) I'm pretty sure it was wrong anyway. (It compares the previous size of the parent to the current size of a child.)